espr/src/ir/type_ref.rs

```diff
@@ -82,6 +82,17 @@ impl TypeRef {
         }
     }
 
+    pub fn to_not_supertype(&self) -> Self {
+        match self {
+            TypeRef::Entity { name, scope, .. } => TypeRef::Entity {
+                name: name.clone(),
+                scope: scope.clone(),
+                is_supertype: false,
+            },
+            _ => self.clone(),
+        }
+    }
+
```

Comment:
The result of this function should not be `TypeRef`. We should create another type for it.